### PR TITLE
Bump go-libp2p

### PR DESCRIPTION
**Pull request by jimmykarily**
Trying to fix tests from this PR: https://github.com/libp2p/go-libp2p-pubsub/pull/584

### CODE REVIEW

**Comment by mudler** [comment]
:wave: @vyzo @marten-seemann maybe you guys can help here in the reviews? context: go1.22 is affected by CVEs (https://osv.dev/vulnerability/GO-2024-3302) https://github.com/quic-go/quic-go/pull/4729 

Thank you in advance ! :bow:

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,15 @@
 module github.com/libp2p/go-libp2p-pubsub
 
-go 1.22
+go 1.22.0
+
+toolchain go1.23.3
```
Comment (marten-seemann) [inline go.mod]: Not necessary.

File: go.mod [hunk 2]
```diff
@@ -1,13 +1,15 @@
 module github.com/libp2p/go-libp2p-pubsub
 
-go 1.22
+go 1.22.0
```
Comment (marten-seemann) [inline go.mod]: Don't change this.

**Review by marten-seemann — CHANGES_REQUESTED** [review]
Misleading PR title. This is NOT fixing a test. It's simply bumping some deps.

**Comment by jimmykarily** [comment]
> Misleading PR title. This is NOT fixing a test. It's simply bumping some deps.

The title refers to the tests failing on this PR: https://github.com/libp2p/go-libp2p-pubsub/pull/584

File: go.mod (hunk 1 above)
Comment (jimmykarily) [reply go.mod]: It was automatically changed because of my local versions used. Removed now.

File: go.mod (hunk 2 above)
Comment (jimmykarily) [reply go.mod]: removed

**Comment by jimmykarily** [comment]
> > Misleading PR title. This is NOT fixing a test. It's simply bumping some deps.
> 
> The title refers to the tests failing on this PR: #584

Reworded the commit message to make more sense after merging

**Review by marten-seemann — APPROVED** [review]

**Comment by vyzo** [comment]
What exactly is the issue?

We usually bump go-libp2p only when there are breaking changes, the idea is not to force any particular version and make sure latest works. We do that as matter of policy because upgrading libp2p in upstream projects is kind of a big deal.

Having said that, if there is a good reason for the bump, sure lets do it.

**Comment by vyzo** [comment]
the right way to do this is with a pr directly to master, dependabot can rebase.

**Comment by marten-seemann** [comment]
The quic-go vulnerability only shows up because quic-go is imported by go-libp2p.

**Comment by jimmykarily** [comment]
> the right way to do this is with a pr directly to master, dependabot can rebase.

I don't see any difference but I can change it if you prefer that

File: go.mod (hunk 2 above)
Comment (jimmykarily) [reply go.mod]: For some reason, without this change, if fails with:

```
go: updates to go.mod needed; to update it:
	go mod tidy
```
strange go...

File: go.mod (hunk 1 above)
Comment (jimmykarily) [reply go.mod]: https://github.com/golang/go/issues/65847

**Comment by jimmykarily** [comment]
> > the right way to do this is with a pr directly to master, dependabot can rebase.
> 
> I don't see any difference but I can change it if you prefer that

done: https://github.com/libp2p/go-libp2p-pubsub/pull/586

**Pull request closed without merging**